setup.py

```diff
@@ -155,9 +155,14 @@
 ]
 extras_require = {
     ':sys_platform != "win32"': ['terminado>=0.3.3'],
-    'doc': ['Sphinx>=1.1'],
+    'doc': [
+        'Sphinx>=1.1',
```

Comment:
For now, I would like to leave the setup.py file untouched and split it out from this PR. I realize that impacts the parts you mention. In the future, as part of a release, we can revisit the setup.py. 

As for the directions, you can comment them out and replace with a simple list of dependencies for the user to install when building docs or running tests or I can merge an updated PR with setup.py removed from the PR and iterate on the document. Either is cool with me.

Thanks! :sunrise: